internal/upgrade/v0_11_0.go

```diff
@@ -0,0 +1,122 @@
+package upgrade
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/grafana/tempo-operator/apis/tempo/v1alpha1"
+	"github.com/grafana/tempo-operator/internal/manifests/manifestutils"
+	"github.com/grafana/tempo-operator/internal/manifests/naming"
+)
+
+// Upstream Tempo 2.5.0 has a breaking change https://github.com/grafana/tempo/releases/tag/v2.5.0
+// The /var/tempo is created in the dockerfile with 10001:10001
+// The user is changed to 10001:10001
+// The previous user in 2.4.2 was root (0)
+// The Red Hat Tempo image does not use root user (it uses 1001) and on OpenShift the /var/tempo PV has a different fsGroup
+// so the issue does not happen on OpenShift.
+func upgrade0_11_0(ctx context.Context, u Upgrade, tempo *v1alpha1.TempoStack) error {
+	// do nothing on OpenShift
+	if u.CtrlConfig.Gates.OpenShift.OpenShiftRoute {
+		return nil
+	}
+
+	image := tempo.Spec.Images.Tempo
+	if image == "" {
+		image = u.CtrlConfig.DefaultImages.Tempo
+	}
+
+	listOps := []client.ListOption{
+		client.MatchingLabels(manifestutils.ComponentLabels(manifestutils.IngesterComponentName, tempo.Name)),
+	}
+	pvcs := &corev1.PersistentVolumeClaimList{}
+	err := u.Client.List(ctx, pvcs, listOps...)
+	if err != nil {
+		return err
+	}
+
+	return upgrade0_11_0_pvcs(ctx, u, tempo.ObjectMeta, tempo.Spec.Template.Ingester.NodeSelector, image, pvcs)
+}
+
+func upgrade0_11_0_monolithic(ctx context.Context, u Upgrade, tempo *v1alpha1.TempoMonolithic) error {
+	// do nothing on OpenShift
+	if u.CtrlConfig.Gates.OpenShift.OpenShiftRoute {
+		return nil
+	}
+
+	listOps := []client.ListOption{
+		client.MatchingLabels(manifestutils.ComponentLabels(manifestutils.TempoMonolithComponentName, tempo.Name)),
+	}
+	pvcs := &corev1.PersistentVolumeClaimList{}
+	err := u.Client.List(ctx, pvcs, listOps...)
+	if err != nil {
+		return err
+	}
+	return upgrade0_11_0_pvcs(ctx, u, tempo.ObjectMeta, tempo.Spec.NodeSelector, u.CtrlConfig.DefaultImages.Tempo, pvcs)
+}
+
+func upgrade0_11_0_pvcs(ctx context.Context, u Upgrade, tempo metav1.ObjectMeta, nodeSelector map[string]string, image string, pvcs *corev1.PersistentVolumeClaimList) error {
+	// keep the jobs around for 1 day
+	ttl := int32(60 * 60 * 24)
```

Comment:
Is this TTL necessary?

Reply:
otherwise the job would hang there forever